core/src/main/java/io/questdb/cairo/vm/MemoryCMARWImpl.java

```diff
@@ -210,10 +210,9 @@ public void switchTo(int fd, long offset, byte truncateMode) {
     }
 
     public void sync(boolean async) {
-        if (pageAddress != 0 && ff.msync(pageAddress, size, async) == 0) {
-            return;
+        if (fd != -1) {
+            ff.fsync(fd);
```

Comment:
nit: A heads-up. Mixing mmap-based writes with fsync may be prone to bugs on certain file systems. OSes other than Linux may also have a hard time with this usage pattern.

The same consideration applies to other similar changes from this PR.

Reply:
fixed just this for now here: https://github.com/questdb/questdb/pull/3304/commits/83f45037735546bc7e05a22925e75d89e408483b